`ioctl_ficlone` is not available on sparc64 linux

```
error[E0425]: cannot find function `ioctl_ficlone` in module `rustix::fs`
  --> /var/tmp/portage/dev-util/maturin-1.12.6/work/maturin-1.12.6/vendor/reflink-copy-0.1.28/src/sys/unix/linux.rs:12:17
   |
12 |     rustix::fs::ioctl_ficlone(&dest, &src)?;
   |                 ^^^^^^^^^^^^^ not found in `rustix::fs`
```

See https://github.com/PyO3/maturin/issues/3065 and https://docs.rs/rustix/latest/rustix/fs/fn.ioctl_ficlone.html

--- a/src/sys/unix/mod.rs
+++ b/src/sys/unix/mod.rs
@@ -1,7 +1,8 @@
 use cfg_if::cfg_if;
 
 cfg_if! {
-    if #[cfg(any(target_os = "linux", target_os = "android"))] {
+    // ioctl_ficlone / FICLONERANGE are not available on SPARC platforms
+    if #[cfg(all(any(target_os = "linux", target_os = "android"), not(any(target_arch = "sparc", target_arch = "sparc64"))))] {
         mod linux;
         pub use linux::reflink;
         pub(crate) use linux::reflink_block;

